### ISSUE TITLE
batch-customer implementation 

### ISSUE THREAD

**Issue by deem0n** [hden/debezium-embedded#5]
Hello, 
I understand that it is early software, so the question is how you see ideal implementation of the batched customer ?

Currently we have very simple one:

```clojure
(defn create-batch-consumer [f]
  (reify EmbeddedEngine$ChangeConsumer
    (handleBatch [_ records committer]
      (f records committer)
      (.markBatchFinished committer))))
```
Comparing to [Java version](https://github.com/debezium/debezium/blob/d3b33e5fbde970b4e1228ddefd608be911f03aac/debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java#L387)
```java
            public void handleBatch(List<SourceRecord> records, RecordCommitter committer) throws InterruptedException {
                try {
                    for (SourceRecord record : records) {
                        try {
                            consumer.accept(record);
                            committer.markProcessed(record);
                        }
                        catch (StopConnectorException ex) {
                            // ensure that we mark the record as finished
                            // in this case
                            committer.markProcessed(record);
                            throw ex;
                        }
                    }
                }
                finally {
                    committer.markBatchFinished();
                }
}
```

So the question is, will we have something similar in Clojure, i.e. with automatic exception handler and automatic `committer.markProcessed(record)`, or alternatively, should I just call Java `ChangeConsumer` directly from my code ? What is your vision and advise ?

**Comment by hden** [comment]
Yes, a better batch consumer would be an excellent idea. However I'm not sure if we should mark the record as processed when an exception is thrown. Maybe a user would like to retry the record when the connector is restarted? Any suggestion?

**Comment by deem0n** [comment]
In the Java example record is marked as processed only in case of `StopConnectorException`. So probably we should just do the same?

**Comment by hden** [comment]
> StopConnectorException  ... this exception should only be raised after that consumer has safely processed the passed event. -- [source](https://github.com/debezium/debezium/blob/v0.9.5.Final/debezium-embedded/src/main/java/io/debezium/embedded/StopConnectorException.java#L13-L15)

You are right. Let's implement that.

- A PR would be super helpful.
- It would be extra helpful if we have a `create-stop-connector-exception` helper function so that Java interop is being taken cared.

**Comment by hden** [comment]
shipped with io.debezium/debezium-embedded 3.0.8
